Make search query in alert conditions configurable in the UI

bernd opened the pull request:
This exposes a new query config field in the alert condition
configuration. The value defaults to "*".

The query is used to build a filter string for the alert condition
search. We add the query string to the query filter for performance
reasons.

This PR also adds two API endpoints to test new and existing alert conditions.

Fixes #3966

**Attention:** This needs to be cherry-picked into 2.5 once merged

kroepke commented on graylog2-server/src/main/java/org/graylog2/alerts/AbstractAlertCondition.java:
```diff
@@ -155,6 +161,25 @@ public boolean shouldRepeatNotifications() {
+
+        final StringBuilder builder = new StringBuilder().append("streams:").append(streamId);
+
+        if (!isNullOrEmpty(query) && !"*".equals(query)) {
```
`CK_QUERY_DEFAULT_VALUE` and maybe trim?

bernd replied:
I don't want to use the configuration constant here because the filter builder would break if the default value changes. I updated the code to trim the streamId and query, though. Good catch, thanks! :+1:

edmundoa commented on graylog2-server/src/main/java/org/graylog2/alerts/AbstractAlertCondition.java:
```diff
@@ -155,6 +160,32 @@ public boolean shouldRepeatNotifications() {
+        if (query != null) {
+            final String trimmedQuery = query.trim();
+            if (!trimmedQuery.isEmpty() && !"*".equals(trimmedQuery)) {
+                builder.append(" ").append(trimmedQuery);
```
Shouldn't we join both filters with an `AND` operator? Otherwise this does not further filter messages in the stream, as the user would expect.

bernd commented:
> Changing the alert condition query to AND POST worked as I initially expected, so to me it looks like we are missing to use the right operator between the stream query filter and the query the user provided.

@edmundoa Yes, you are right. I rebased and updated the PR to use the `AND` operator to combine the streams query and the user defined query.
